django/db/models/sql/query.py

```diff
@@ -534,7 +534,12 @@ def get_aggregation(self, using, aggregate_exprs):
             # Queries with distinct_fields need ordering and when a limit is
             # applied we must take the slice from the ordered query. Otherwise
             # no need for ordering.
-            inner_query.clear_ordering(force=False)
+            if (
+                not inner_query.is_sliced
+                and not inner_query.distinct_fields
+                and inner_query.orderby_issubset_groupby
+            ):
+                inner_query.clear_ordering(force=True)
```

Comment:
Github chewed up my last comment, leaving it again.

Is there reason you didn't opt for:
```suggestion
            if inner_query.orderby_issubset_groupby:
                inner_query.clear_ordering(force=False)
```

Reply:
I followed blindly the comments (line: 534) hoping that is true:
```python
# Queries with distinct_fields need ordering and when a limit is
# applied we must take the slice from the ordered query. Otherwise
# no need for ordering.
```

Reply:
For sure, I just mean the `clear_ordering()` actually handles that for you and also checks `select_for_update`. See the beginning of `clear_ordering()`.

Just so that we don't actually introduce any drift against that implementation, I'll roll this in for you 👍